### ISSUE TITLE
`systemd::journal` doesn't return Result

### ISSUE THREAD

**Issue by cglong** [jmesmon/rust-systemd#304]
We were looking at using `systemd::journal` for sending messages to journald, but noticed [all the functions](https://docs.rs/systemd/latest/systemd/journal/index.html#functions) either return a `c_int` or nothing at all. Given the `c_int` values aren't well-defined in the docs, we essentially have no way of knowing whether the message was successfully sent or not.

Could these functions be updated to instead return a `Result` or, less ideally, could the docs be updated to include details on how to interpret the `c_int`s? Thank you!

**Comment by codyps** [comment]
I agree that these should return a `Result` type instead of a plain `c_int`. Would happily take a patch to do this.
